lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -933,6 +932,13 @@ public Executor getExecutor() {
     return executor;
   }
 
+  /**
+   * Returns this searchers slice execution control plane or <code>null</code> if no executor was provided
+   */
+  public SliceExecutor getSliceExecutor() {
```

Comment:
we shouldn't make this method public if it returns a pkg-private class, let's make the method pkg-private too? Or even remove it entirely as I'm not seeing any call site for it?